src/app/AlwaysOn.Shared/TelemetryExtensions/AppInsightsRequestHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Azure.Cosmos;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AlwaysOn.Shared.TelemetryExtensions
+{
+    public class AppInsightsRequestHandler : RequestHandler
+    {
+        public AppInsightsRequestHandler(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient;
+        }
+
+        private readonly TelemetryClient _telemetryClient;
+
+        public override async Task<ResponseMessage> SendAsync(RequestMessage request, CancellationToken cancellationToken)
+        {
+            var title = request.Properties.ContainsKey("Operation") ? request.Properties["Operation"].ToString() : $"{request.Method} {request.RequestUri.OriginalString}";
+            var dbClientEndpoint = request.Properties.ContainsKey("DbClientEndpoint") ? request.Properties["DbClientEndpoint"].ToString() : null;
```

Comment:
doesnt this work?

    request.Properties["DbClientEndpoint"]?.ToString()

Reply:
Nope -> `KeyNotFoundException`.

![image](https://user-images.githubusercontent.com/9624293/183371347-84a5d1ba-e360-4d7b-b5f7-f39db92767c5.png)

Reply:
ah too bad. I thought that would work